src/hotspot/share/opto/vectornode.cpp

```diff
@@ -135,7 +135,14 @@ int VectorNode::opcode(int sopc, BasicType bt) {
   case Op_AbsD:
     return (bt == T_DOUBLE ? Op_AbsVD : 0);
   case Op_NegI:
-    return (bt == T_INT ? Op_NegVI : 0);
+    switch (bt) {
+      case T_BYTE:
+      case T_SHORT:
+      case T_INT: return Op_NegVI;
```

Comment:
Why not add `Op_NegVB` for `BYTE` and `Op_NegVS` for `SHORT`?
Is there any performance drop for byte/short negation operation if both of them are handled as a NegVI vector?

Reply:
> The compiler can get the real type info from `Op_NegVI` that can also handle the `BYTE ` and `SHORT ` basic type. I just don't want to add more new IRs which also need more match rules in the ad files.
> 
> > Is there any performance drop for byte/short negation operation if both of them are handled as a NegVI vector?
> 
> From the benchmark results I showed in the commit message, I didn't see not any performance drop for byte/short. Thanks!

There seems no vectorized instructions for {byte, short, int, long} on x86, so this should be fine on x86.
I tested the patch on x86 and the performance number looks good.

Reply:
Thanks for doing this! Yeah, I think the performance for masked negation operations might improve on non avx-512 systems.